pkg/controller/nodemaintenance/check_lease_update.go

```diff
@@ -0,0 +1,431 @@
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{ {
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+						} },
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          getTimeLease(int64(NowTime)),
+						RenewTime:            getTimeLease(int64(NowTime)),
+						LeaseTransitions:     makeInt32(4),
+					},
+				},
+			 ),
+
+		Entry("extend lease if same holder and expired lease - check modified lease duration",
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: "v1",
+								Kind:       "Node",
+								Name:       "@",
+								UID:        "#",
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds-42),
+						AcquireTime:          nil,
+						RenewTime:            getTimeLease(int64(NowTime-LeaseDurationInSeconds-1)),
+						LeaseTransitions:     nil,
+						},
+				},
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{ {
+								APIVersion: corev1.SchemeGroupVersion.WithKind("Node").Version,
+								Kind:       corev1.SchemeGroupVersion.WithKind("Node").Kind,
+								Name:       getMockNode().Name,
+								UID:        getMockNode().UID,
+						} },
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          getTimeLease(int64(NowTime)),
+						RenewTime:            getTimeLease(int64(NowTime)),
+						LeaseTransitions:     makeInt32(1),
+					},
+				},
+			 ),
+
+		Entry("extend lease if same holder and expired lease",
+				&coordv1beta1.Lease{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: "v1",
+								Kind:       "Node",
+								Name:       "@",
+								UID:        "#",
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString(LeaseHolderIdentity),
+						LeaseDurationSeconds: makeInt32(LeaseDurationInSeconds),
+						AcquireTime:          nil,
+						RenewTime:            getTimeLease(int64(NowTime-LeaseDurationInSeconds-1)),
+						LeaseTransitions:     nil,
```

Comment:
If we're extending our own lease, then `AcquireTime` and `LeaseTransitions` would never be `nil`

Reply:
Ok, fixe that. go mixed up because of your previous comment "If the lease has previously expired, then we are acquiring it now and AcquireTime should reflect that if it was previously nil". Now added another test case to differentiate that; we have now extend lease if same holder and expired lease (acquire time previously not nil)" and "extend lease if same holder and expired lease (acquire time previously nil)"